### ISSUE TITLE
Output of has_properties over verbose

### ISSUE THREAD

**Issue by keis** [hamcrest/PyHamcrest#72]
I think the mismatch description of has_porperties is very verbose and actually making readability worse.

```
assert_that(b, has_properties(x=6))
Expected: (an object with a property 'x' matching <6>)
     but: an object with a property 'x' matching <6> property 'x' was <7>
```

this is from all_of repeating the offending matcher, which does make sense when the user composes various matchers. But has_properties to the user is a single matcher and the output is only confusing if you don't know it's made up of multiple matchers.

**Comment by offbyone** [comment]
Agreed. It's one of my least favorite outputs here. I'd love to fix it, but I'd love a pull request more :)

(translation: currently on low ebb re: open source work, feel free to propose a fix)

**Comment by brunns** [comment]
What would we like the mismatch description to read?

**Comment by brunns** [comment]
This might work, assuming `but: property 'x' was <7>` is what you're looking for.

**Comment by keis** [comment]
Yeah, I think that's much better. The description itself can be very repetitive too. IIRC in the code base that prompted me to open this we would regularly use has_properties with 10-12 properties at once, and sometimes nested, which ended up with the phrase "an object with a property" repeated over and over.

For a matcher `has_properties(x=6, y=5, z=7)` the description becomes `(an object with a property 'x' matching <6> and an object with a property 'y' matching <5> and an object with a property 'z' matching <7>)` while I would ideally see something like

`an object with properties 'x' matching <6>, and 'y' matching <5>, and 'z' matching <7>`

**Comment by brunns** [comment]
Right now, `all_of()` reports only the first failing matcher. Might we want to change that? Only when used by `has_properties()`, or all the time? (Doing it all the time might be a breaking change for some users, I think.)

**Comment by offbyone** [comment]
Can you update that PR with comparison outputs please?

**Comment by brunns** [comment]
Sorry @offbyone, not sure I follow. What's the change you'd like?

**Comment by brunns** [comment]
@keis I like the improved description idea a lot, but I think I would do it as a separate PR.

**Comment by offbyone** [comment]
I'm having a hard time visualizing what this change would look like on a complex match. Can you include, either in the commit message or in the PR itself, a concrete example of what changed?

**Comment by brunns** [comment]
@offbyone Gotcha. I've added just such a concrete example!

**Comment by brunns** [comment]
Should be fixed in release 1.10.
